avocado/core/multiplexer.py

```diff
@@ -408,8 +409,10 @@ def get_number_of_tests(self, test_suite):
         """
         # Currently number of tests is symetrical
         if self.variants:
-            return (len(test_suite) *
-                    sum(1 for _ in self.variants))
+            no_variants = sum(1 for _ in self.variants)
```

Comment:
Well this solution is from the internet :smile:, they say it's the fastest low-memory-footprint way to count items.